src/main/java/org/sciborgs1155/robot/Hopper/Hopper.java

```diff
@@ -0,0 +1,54 @@
+package org.sciborgs1155.robot.Hopper;
+
+import static org.sciborgs1155.robot.Hopper.HopperConstants.*;
+
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.Commands;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import monologue.Logged;
+import monologue.Monologue.LogBoth;
+import org.sciborgs1155.robot.Robot;
+
+public class Hopper extends SubsystemBase implements Logged {
+
+  final PIDController pid = new PIDController(kp, ki, kd);
+  public final HopperIO hopper = Robot.isReal() ? new RealHopper() : new SimHopper();
+
+  @LogBoth public double targetSpeed = pid.getSetpoint();
+
+  @LogBoth
+  public double speed() {
+    return hopper.getSpeed();
+  }
+
+  @LogBoth private boolean isAtTarget = pid.atSetpoint();
+
+  @Override
+  public void periodic() {
+    hopper.setVoltage(pid.calculate(hopper.getSpeed(), targetSpeed));
```

Comment:
so you don't actually need to have a targetSpeed field at all. You can just get rid of the middle man. You're setting the pidSetpoint with your forward and backward commands, so you don't need to give it a new setpoint here -- it's already set. You can just do pid.calculate(hopper.getSpeed()) instead. The one catch with this is that you should make sure that the initial setpoint is 0 (you can just set that in your constructor)